### PR TITLE
Align deployment image name when custom name is set in Openshift configuration

### LINKED ISSUE #32439: Openshift resources getting out of sync when custom quarkus.openshift.name is used.
### Describe the bug

It seems that we also need to align the `quarkus.container-image.name` to match.
More specifically the issue as explained to me was worded: 

`If I choose to override the application name from the pom artefact default using quarkus.openshift.name, then I absolutely have to also set the quarkus.container-image.name to match otherwise I get a Deployment detached from the image that is pushed.`

### How to reproduce

1. Create a quarkus app with the tool of your choice.
2. Add the `quarkus-openshift` extension to the project.
3. Use `Deploymnet` kind: `quarkus.openshift.deployment-kind=Deployment`.
4. Use a custom name of the manifests: `quarkus.openshift.name=custom-name` 

### Expected behavior

The app should be deployed and working

### Actual behavior
App not starting due to `ImagePullBackOff`.
With closer inspection we see that the image used makes use of the original name and not the custom one, while the image streams do use the custom one.

### PR DESCRIPTION
**Pull request by iocanel**
Resolves: #32439

### CODE REVIEW

**Comment by geoand** [comment]
Please update the title :)

File: extensions/container-image/spi/src/main/java/io/quarkus/container/spi/ContainerImageCustomNameBuildItem.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package io.quarkus.container.spi;
+
+import io.quarkus.builder.item.BuildItem;
+import io.quarkus.builder.item.SimpleBuildItem;
+
+/**
+ * This {@link BuildItem} can be used to override the default image name.
+ * It can be used in cases where the name of the image is customized externally.
+ * Example: The openshift extension may override the name. To ensure that things are in symc
```
Comment (geoand) [inline extensions/container-image/spi/src/main/java/io/quarkus/container/spi/ContainerImageCustomNameBuildItem.java:9]:
```suggestion
 * Example: The openshift extension may override the name. To ensure that things are in sync
```

File: extensions/container-image/spi/src/main/java/io/quarkus/container/spi/ContainerImageCustomNameBuildItem.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * the image name needs to be set.
```
Comment (geoand) [inline extensions/container-image/spi/src/main/java/io/quarkus/container/spi/ContainerImageCustomNameBuildItem.java:10]:
```suggestion
 * with the image name needs to be set.
```

**Comment by iocanel** [comment]
@geoand: committed suggestions. Wanna have a second look?

**Comment by geoand** [comment]
Please squash the commits

**Comment by geoand** [comment]
LGTM otherwise

**Comment by gsmet** [comment]
I squashed the commits.

**Review by Sgitario — APPROVED** [review]

**Pull request merged**